CondCore/JetMETPlugins/plugins/JetResolution_PayloadInspector.cc

```diff
@@ -0,0 +1,392 @@
+        }
+      }
+      return false;
+    }
+  };  // class
+
+  class JetResolutionVsPt : public cond::payloadInspector::Histogram1D<JetResolutionObject, SINGLE_IOV> {
+  public:
+    static const int MIN_PT = -0.0;
+    static const int MAX_PT = 3000.0;
+    static const int NBIN = 300;
+
+    JetResolutionVsPt()
+        : cond::payloadInspector::Histogram1D<JetResolutionObject, SINGLE_IOV>(
+              "Jet Energy Resolution", "p_T", NBIN, MIN_PT, MAX_PT, "Resolution") {}
+
+    bool fill() override {
+      auto tag = PlotBase::getTag<0>();
+      for (auto const& iov : tag.iovs) {
+        std::shared_ptr<JetResolutionObject> payload = Base::fetchPayload(std::get<1>(iov));
+        if (payload.get()) {
+          if (!payload->getRecords().empty() &&  // No formula for SF
+              payload->getDefinition().getFormulaString().compare("") == 0)
+            return false;
+
+          for (const auto& record : payload->getRecords()) {
+            // Check Eta & Rho
+            if (!record.getBinsRange().empty() && payload->getDefinition().getBinName(0) == "JetEta" &&
+                record.getBinsRange()[0].is_inside(2.30)) {
+              if (record.getBinsRange().size() > 1 && payload->getDefinition().getBinName(1) == "Rho" &&
+                  record.getBinsRange()[1].is_inside(15.)) {
+                if (!record.getVariablesRange().empty() && payload->getDefinition().getVariableName(0) == "JetPt") {
+                  reco::FormulaEvaluator f(payload->getDefinition().getFormulaString());
+
+                  for (size_t idx = 0; idx <= NBIN; idx++) {
+                    double x_axis = (idx + 0.5) * (MAX_PT - MIN_PT) / NBIN + MIN_PT;
+                    if (record.getVariablesRange()[0].is_inside(x_axis)) {
+                      std::vector<double> var = {x_axis};
+                      std::vector<double> param;
+                      for (size_t i = 0; i < record.getParametersValues().size(); i++) {
+                        double par = record.getParametersValues()[i];
+                        param.push_back(par);
+                      }
+                      float res = f.evaluate(var, param);
+                      fillWithBinAndValue(idx + 1, res);
+                    }
+                  }
+                }
+              }
+            }
+          }
+          return true;
+        }
+      }
+      return false;
+    }
+  };
+
+  class JetResolutionSummary : public cond::payloadInspector::PlotImage<JetResolutionObject> {
+  public:
+    static const int MIN_ETA = -5.05;
+    static const int MAX_ETA = 5.05;
+    static const int NBIN_ETA = 51;
+    static const int MIN_PT = -5.0;
+    static const int MAX_PT = 3005.0;
+    static const int NBIN_PT = 302;
+
+    JetResolutionSummary() : cond::payloadInspector::PlotImage<JetResolutionObject>("Jet Resolution Summary") {
+      Base::setSingleIov(true);
+    }
+
+    bool fill(const std::vector<std::tuple<cond::Time_t, cond::Hash> >& iovs) override {
```

Comment:
you need to use the `SINGLE_IOV` template in order to use the other one.

Reply:
> JetResolution_PayloadInspector.cc: In constructor 'JME::JetScaleFactorSummary::JetScaleFactorSummary()':
> JetResolution_PayloadInspector.cc:308:55: error: type 'cond::payloadInspector::PlotImage<JME::JetResolutionObject, cond::payloadInspector::SINGLE_IOV>' is not a direct base of 'JME::JetScaleFactorSummary'
>   308 |     JetScaleFactorSummary() : cond::payloadInspector::PlotImage<JetResolutionObject, SINGLE_IOV>("Jet ScaleFactor Summary") {
>       |                                                       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
> JetResolution_PayloadInspector.cc:308:123: error: no matching function for call to 'cond::payloadInspector::PlotImage<JME::JetResolutionObject>::PlotImage()'
>   308 | otImage<JetResolutionObject, SINGLE_IOV>("Jet ScaleFactor Summary") {
>       |                                                                   ^
> 
Looks there is no template with SINGLE_IOV for PlotImage class.

Reply:
@yuanchao 

> Looks there is no template with SINGLE_IOV for PlotImage class.

this is simply not true:

https://github.com/cms-sw/cmssw/blob/166eab1458287d877b7b06a931153f21b74e2093/CondCore/SiPixelPlugins/plugins/SiPixelLorentzAngle_PayloadInspector.cc#L69-L71

Reply:
Emm.. my bad. I missed one declaration. There should be some other issue previously, anyway... I should have followed SiPixel PI module as example. ;) These parts are now fixed.

Reply:
thanks for the updates. It would be nice to to have a unit test as for the other PI packages:

https://github.com/cms-sw/cmssw/blob/166eab1458287d877b7b06a931153f21b74e2093/CondCore/AlignmentPlugins/test/BuildFile.xml#L1-L6

https://github.com/cms-sw/cmssw/blob/166eab1458287d877b7b06a931153f21b74e2093/CondCore/AlignmentPlugins/test/testTrackerAlignmentPayloadInspector.cpp#L1-L10

but I let conveners decide if that's mandatory or not.